Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs

```diff
@@ -35,13 +37,19 @@ public class LogExceptionAnalyzer : DiagnosticAnalyzer
 				description: Description
 			);
 
-		private string _logMethodNames;
-
+		private IEnumerable<string> _logMethodNames;

```

Comment:
There is a compiler error built into the diagnostic analyzers for not storing per-compilation data in the fields of the analyzer.  I think this technically qualifies as per-compilation?

Also, I would type this as a hashset, not an IEnumerable.

Reply:
Updated to not store per-compilation data.  Good catch.  Re hashset, that's an improvement for AdditionalFilesHelper that I'm going to declare as outside the scope of this PR as it affects several analyzers.